tcconfig/traffic_control.py

```diff
@@ -478,6 +478,7 @@ def __delete_ingress_qdisc(self):
                             "RTNETLINK answers: Invalid argument",  # debian/ubuntu
                             "RTNETLINK answers: No such file or directory",  # debian/ubuntu
                             "Error: Cannot find specified qdisc on specified device",  # RHEL/fedora
+                            "Error: Invalid handle.",  # debian/ubuntu
```

Comment:
I am not sure when the error message is returned from `tc` command.

Could you possibly describe steps to reproduce and the environments?

Reply:
I hit this on Debian 9 and 10
When running `tcdel veth3 --all` multiple times (when there were no rules on the interface) I would hit this.

Reply:
I have got the following error message with `debian 10` `iproute2 4.20.0-2`:

```
Error: Cannot delete qdisc with handle of zero.
```

what exactly your environments?

Reply:
I have the same version of iproute2.
Here is the error I got:
```
[00:07:17.960][info] [WARNING] command='tcdel wlan0 --all', returncode=1, stderr="\x1b[33m\x1b[1m[WARNING]\x1b[0m command='/sbin/tc qdisc del dev wlan0 root', returncode=2, stderr='Error: Cannot delete qdisc with handle of zero.\\n'\n\x1b[31m\x1b[1m[ERROR]\x1b[0m command execution failed\n  command=/sbin/tc qdisc del dev wlan0 ingress\n  stderr=Error: Invalid handle.\n\n\x1b[1m[INFO]\x1b[0m delete wlan0 ifb device (ifb5898)\n"
```

Here is the output cleaned up a bit:
```
[00:07:17.960][info] [WARNING] command='tcdel wlan0 --all', returncode=1, stderr="\x1b[33m\x1b[1m[WARNING]\x1b[0m command='/sbin/tc qdisc del dev wlan0 root', returncode=2, stderr='Error: Cannot delete qdisc with handle of zero.\
'
[ERROR] command execution failed
  command=/sbin/tc qdisc del dev wlan0 ingress
  stderr=Error: Invalid handle.

[INFO] delete wlan0 ifb device (ifb5898)
"
```

Reply:
In that case, the difference of the error message would stem from OS.
Please replace the comment (`# debian/ubuntu`) to a specific OS with version.

Reply:
We are running Debian 10 Buster, with a slightly customized 5.7.8 kernel (nothing touched in the networking area) so I believe this should be reproducible on vanilla Debian (and therefore likely Ubuntu).

I don't get that error every time, only sometimes and I haven't figured out a reliable reproducible case yet, but I do believe the comment to be accurate.

Reply:
Thank you for your additional information.

If the error conditions are not clear, I'm not so sure whether the error is ignorable or not.